.github/workflows/_integration-tests.yml

```diff
@@ -89,12 +97,13 @@ jobs:
         -   uses: actions/setup-python@v5
```

Comment:
do we even need to do `actions/setup-python@v5` per-python version? Pinning to a single python version might save time in GHA being able to cache more intelligently

Reply:
I don't follow. You only want to run integration tests against one version of Python everywhere? We already limit PR checks to one version and leave it to scheduled tests for checking all versions. This option would need to be here to support the latter.